tools/cldr/cldr-to-icu/build-icu-data.xml

```diff
@@ -15,7 +15,7 @@
       inconsistent state when the conversion is cancelled). -->
 <project name="Convert" default="all" basedir=".">
 
-    <target name="all" depends="init-args, prepare-jar, clean, convert"/>
+    <target name="all" depends="init-args, prepare-jar, clean, convert, clean-cldrtestdata, copy-cldrtestdata"/>
```

Comment:
Why does "conversion" need test data at all? I really don't understand this.

Reply:
unitsTest.txt:
- Tests that conversions that should be possible on account of CLDR conversion data are indeed possible in the ICU implementation. (In theory this could be used for other implementations too.) Converting anything to anything that is compatible requires a combination of the right conversion factors.
- Can also be used to test precision: the ICU implementation is not at "arbitrary precision" at the moment, we can increase test accuracy and use these tests to ensure we are doing arbitrary precision. (In the constants used in these conversions, I think it's only "PI" and "G" that are approximate, the rest should be accurate to the appropriate number of decimal digits.)

unitPreferencesTest.txt:
An integration test that confirms we're also finding the right localized preferences for the specified usages.

As new units, conversion rates, and unit preferences are added to CLDR, adding them to the CLDR-defined data-driven test suite can check, end-to-end, that the desired output is produced.

Reply:
So is this "test data" we're talking about "data for unit/regression tests" or something else?

Data for unit/regression tests shouldn't be needed when running the tool normally.
There should be a separate test target of some kind for running tests separately.

Reply:
It is a painful mix of both, so I found myself trying to go both ways:

- If CLDR unit preference data is changed, the tests will change (e.g. we find an incorrect preference and want to change its thresholds, or something else, in CLDR). For this, we want the test data in sync with the tests.

- If new preferences are added though, without impact on existing test cases, and there's a bug in ICU, or something missing, we would perhaps *not* want to update the tests too, as it would cause ICU test breakage upon CLDR data import, which the importer would not want to care about? (An example: suppose we add compound units inside mixed units in CDLR, when in ICU the mixed units code only supports built-in units.)

Desireable action, as I see it currently: file a ticket for the broken tests, and comment them out of the test data files?

@macchiati do you have an opinion on handling the CLDR test data for this kind of ICU-impacting "breaking" change?

Reply:
I have started an email thread for this, with subject line "ICU PR 1199 and CLDR Test Data". I'm considering this resolved for now, so I'm squashing the commits again.